lib/Echidna/Campaign.hs

```diff
@@ -235,14 +236,30 @@ addToCorpus res = unless (null rtxs) $ hasLens . corpus %= (rtxs:) where
 randseq :: ( MonadCatch m, MonadRandom m, MonadReader x m, MonadState y m
            , Has GenDict y, Has TxConf x, Has TestConf x, Has CampaignConf x, Has Campaign y)
         => Int -> Map Addr Contract -> World -> m [Tx]
-randseq ql o w = do 
+randseq ql o w = do
   ca <- use hasLens
-  let txs = ca ^. corpus  
-      p   = ca ^. ncallseqs 
-  if length txs > p then -- Replay the transactions in the corpus, if we are executing the first iterations
-    return $ txs !! p
-  else                   -- Randomly generate new transactions 
-    replicateM ql (evalStateT (genTxM o) (w, ca ^. genDict))
+  let ctxs = ca ^. corpus
+      p    = ca ^. ncallseqs
+  if length ctxs > p then -- Replay the transactions in the corpus, if we are executing the first iterations
+    return $ ctxs !! p
+  else
+    do
+      -- Randomly generate new random transactions
+      gtxs <- replicateM ql (evalStateT (genTxM o) (w, ca ^. genDict))
+      n <- getRandomR (0, 2 :: Integer)
+      case (n, ctxs) of
+        -- Use the generated random transactions
+        (_, []) -> return gtxs
+        (0, _ ) -> return gtxs
+        -- Append a sequence from the corpus with random ones
+        (1, _ ) -> do rtxs <- rElem $ NE.fromList ctxs
+                      k <- getRandomR (0, length rtxs - 1 )
+                      return $ take ql $ take k rtxs ++ gtxs
+        -- Prepend a sequence from the corpus with random ones
+        (2, _ ) -> do rtxs <- rElem $ NE.fromList ctxs
+                      k <- getRandomR (0, length rtxs - 1 )
+                      return $ take ql $ take k gtxs ++ rtxs
+        _       -> error "invalid pattern in randseq"
```

Comment:
we should properly define functions and then take a weighted random selection instead of hardcoding some integers.

Reply:
also instead of `(0, _ ) ->` use `(0, _)  ->`

Reply:
Ok done. Can you re-check the code?